### PR TITLE
add clarabel

### PR DESCRIPTION
**Pull request by sbarratt**
this passes the tests. i had to lower the atol in the exp_cone test a bit; might be worth looking into.

### CODE REVIEW

**Comment by PTNobel** [comment]
Sorry lower `atol` as in make it smaller? So we're requiring a more accurate answer?

**Comment by sbarratt** [comment]
The opposite. atol is higher. i can look into it later.

**Comment by sbarratt** [comment]
I can't figure it out. did some basic checking, but Clarabel gets a better solution than SCS on both the initial solve and also the perturbed solve, so Im king of scratching my head. @bstellato

**Comment by bungun** [comment]
How big is the kingdom of head scratching?On 1 Sep 2023, at 00:06, Shane Barratt ***@***.***> wrote:﻿
I can't figure it out. did some basic checking, but Clarabel gets a better solution than SCS on both the initial solve and also the perturbed solve, so Im king of scratching my head. @bstellato

**Comment by bstellato** [comment]
Could it be an issue of the linear system solved to take the derivative? 

Clarabel returns a higher accuracy solution with smaller complementary slackness residuals. It may cause the linear system to be badly scaled and lsqr to return less accurate derivatives. Just a guess though...

**Comment by sbarratt** [comment]
I tried with a variety of backward methods (lsmr, lsqr and dense) and had
similar results

Shane

**Comment by bstellato** [comment]
There were two issues with Clarabel
- It was not using the settings passed
- Its equilibration was making it way less precise

It now works! @phschiele @akshayka @sbarratt what do you think?

**Comment by sbarratt** [comment]
If it now passes tests then go ahead!

File: tests/test_scs.py
Comment (phschiele) [inline tests/test_scs.py:14]:
```suggestion
```

File: tests/test_scs.py
Comment (phschiele) [inline tests/test_scs.py:129]:
```suggestion
```

**Review by phschiele — COMMENTED** [review]
@sbarratt @bstellato Thanks for this PR! I think it is really important to make sure that the main cvxpy extensions support Clarabel to foster adoption in the community. 

I suggest to remove two small debug statements introduced during development.

Could it make sense to disable `equilibrate` by default if it was causing such trouble in the tests?

**Review by phschiele — APPROVED** [review]

**Comment by bstellato** [comment]
Thanks @phschiele. I have removed the print comments (sorry I did not see them before) and reverted the equilibration settings to its default ones. It now works as any other solver, I would not disable equilibration by default. It should be good to go now.

**Comment by phschiele** [comment]
@bstellato Thanks! Let's wait for @akshayka 's review. I'm updating the build in #63, so we can go ahead an cut a release once this is merged.

**Review by akshayka — APPROVED** [review]

**Pull request merged**
